BCStudentSoftwareDevTeam/lsf
# Allow Supervisors Limited Student and Form Search Capabilities

Issue #366 sub-issue of #359 Epic.

## Changes
- Allowed supervisors access to the form search and student search pages and moved those out of the administrator's dropdown menu.
- Created a new logic file which houses student search logic and created the function, limitSearch, which takes in a list of users that were going to be displayed for the student search, and filtered out all students that the current user does not have access to.
- Tested limitSearch in new test file test_search.
- Adjusted queries and conditionals in the main/formSearch.py controller which made it so a user can only search for forms in their department and with other supervisors or students in those departments.

## How to Test
- As an admin, click on the form search and student search pages and confirm that the administrators can see all students, departments and supervisors.
- Then switch to a supervisor with one to a few departments (bolsters is a good one as she just is attached to math) and confirm that their search results are limited only to the scope of their department.
- Run tests.
